api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+
+    properties:
+      data_type:
+        type: string
+        example: 'pennybags:order'
+      data_op:
+        type: string
+        enum: ['C', 'U', 'D', 'S']
+        description: |
+          The type of operation executed on the entity.
+          * C: Creation
+          * U: Update
+          * D: Deletion
+          * S: Snapshot
+    required:
+      - data_type
+      - data_op
+
+  DataChangeEvent:
+    type: object
+    description: |
+      A Data change Event.
+
+      Represents a change on a resource.
+    allOf:
+      - $ref: '#/definitions/DataChangeEventQualifier'
+      - $ref: '#/definitions/EventMetadata'
+    properties:
+      data:
+        type: object
+        description: |
+            The payload of the type
+    required:
+        - data
+    additionalProperties: false
+
+  Problem:
+    type: object
+    properties:
+      type:
+        type: string
+        format: uri
+        description: |
+          An absolute URI that identifies the problem type.  When dereferenced,
+          it SHOULD provide human-readable API documentation for the problem type
+          (e.g., using HTML).
+          This Problem object is the same as provided by https://github.com/zalando/problem
+        example: http://httpstatus.es/503
+      title:
+        type: string
+        description: |
+          A short, summary of the problem type. Written in english and readable
+          for engineers (usually not suited for non technical stakeholders and
+          not localized); example: Service Unavailable
+      status:
+        type: integer
+        format: int32
+        description: |
+          The HTTP status code generated by the origin server for this occurrence
+          of the problem.
+        example: 503
+      detail:
+        type: string
+        description: |
+          A human readable explanation specific to this occurrence of the
+          problem.
+        example: Connection to database timed out
+      instance:
+        type: string
+        format: uri
+        description: |
+          An absolute URI that identifies the specific occurrence of the problem.
+          It may or may not yield further information if dereferenced.
+    required:
+      - type
+      - title
+      - status
+
+  Metrics:
+    type: object
+
+  Topic:
+    description: |
+      Representation of the underlying (implementation dependent) event ordering and persistence structure.
+
+      Direct operation on this is usually unnecessary.
+    type: object
+    required:
+      - name
+    properties:
+      name:
+        type: string
+        description: Topic name
+        example: article.120
+
+  Partition:
+    required:
+      - partition
+      - oldest_available_offset
+      - newest_available_offset
+    properties:
+      partition:
+        type: string
+      oldest_available_offset:
+        type: string
+      newest_available_offset:
+        type: string
```

Comment:
just make clear:  
- oldest_available_offset point to the oldest event in the partition (queue)
- newest_available_offset point to the most recently produced event in the partition (queue)
Both are partition "storage" cursors -- not related related to consumer cursors that will be supported in future HL API extension.